core/src/databases/table_schema.rs

```diff
@@ -161,9 +168,17 @@ impl TableSchema {
                         }
                     }
                     Value::String(_) => TableSchemaFieldType::Text,
-                    Value::Object(_) | Value::Array(_) => Err(anyhow!(
+                    Value::Object(obj) => match Self::try_parse_date_object(obj) {
+                        Some(_) => TableSchemaFieldType::DateTime,
+                        None => Err(anyhow!(
+                            "Field {} is an unsupported object type on row {}",
```

Comment:
Let's align the two error message. I think I like the one below better. Let's maybe go for:

`Field {} is not a primitive or datetime object type on row {} (non datetime object and arrays are not supported)`

Reply:
sgtm